### ISSUE TITLE
how can i use this with my own data

### ISSUE THREAD

**Issue by mithils3** [wangermeng2021/SnapMix-tensorflow2#1]

**Comment by wangermeng2021** [comment]
I am working on it，it will take about 2 or 3 days. if you can't wait, you could modify generator/cub_generator.py

**Comment by mithils3** [comment]
I can wait for a few days

**Comment by wangermeng2021** [comment]
I have done it， you can try it on your own dataset, Let me know if there is a problem.

**Comment by mithils3** [comment]
Can you implement it in something like IDG.flow_from_dataframe

**Comment by mithils3** [comment]
Also a bit a of typo in the repo in the result section it is written GTX2080 but their is no GTX2080 it is RTX2080

**Comment by mithils3** [comment]
Can you please tell me all the arguments of the train.py file

**Comment by wangermeng2021** [comment]
Can you explain more about "IDG.flow_from_dataframe" ?  
You can find all arguments in  train.py file:
"
    parser.add_argument('--epochs', default=300, type=int)
    parser.add_argument('--start-val-epoch', default=100, type=int)
    parser.add_argument('--batch-size', default=16, type=int)
    parser.add_argument('--dataset', default='custom', type=str, help="choices=['cub','cars','custom']")
    parser.add_argument('--dataset-dir', default='dataset/cat_dog', type=str, help="choices=['dataset/cub','dataset/cars','custom_dataset_dir']")
    parser.add_argument('--augment', default='snapmix', type=str, help="choices=['baseline','cutmix','snapmix']")
    parser.add_argument('--model', default='ResNet50', type=str, help="choices=['ResNet50','ResNet101','EfficientNetB0']")
    parser.add_argument('--pretrain', default='imagenet', help="choices=[None,'imagenet','resnet50_weights_tf_dim_ordering_tf_kernels_notop.h5']")
    parser.add_argument('--concat-max-and-average-pool', default=False, type=bool,help="Use concat_max_and_average_pool layer in model")
    parser.add_argument('--lr-scheduler', default='warmup_cosinedecay', type=str, help="choices=['step','warmup_cosinedecay']")
    parser.add_argument('--init-lr', default=1e-3, type=float)
    parser.add_argument('--lr-decay', default=0.1, type=float)
    parser.add_argument('--lr-decay-epoch', default=[80, 150, 180], type=int)
    parser.add_argument('--warmup-lr', default=1e-4, type=float)
    parser.add_argument('--warmup-epochs', default=0, type=int)
    parser.add_argument('--weight-decay', default=1e-4, type=float)
    parser.add_argument('--optimizer', default='sgd', help="choices=['adam','sgd']")
"

**Comment by mithils3** [comment]
Ok so no need of just flow from dataframe method https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image/ImageDataGenerator?version=nightly#flow_from_dataframe here. But I cant use it with a custom model. Right now i can use  only the models present in the models folder. I can contribute a few models. Can you please implement those ?

**Comment by wangermeng2021** [comment]
I have difficulty understanding your question, Could you describe it in more detail?  you want me to implement some other models?(in addition to resnet,efficientnet), if so, could you tell me the model's name.

**Comment by mithils3** [comment]
I mean you could add efficient net with noisy student weights and Vision Information transformers,seresnext etc. If you want to the implementation just tell me

**Comment by wangermeng2021** [comment]
That would be great. noisy student and transformers is very promising.  I am working on another  project about object detection, so I don't have much time to implement it. if you can do it ,that will be great.

**Comment by mithils3** [comment]
Any tips on which file should i add it in and where to add args ?

**Comment by wangermeng2021** [comment]
Sorry for late replay, Can you help me to implement visual transformer?   you can create a  file named  "resnet_vt.py"  under 'model/' , the inputs and outputs of resnet_vt.py keep consistent with "resnet.py"

**Comment by mithils3** [comment]
>Here is a implementation of vit for you https://pypi.org/project/vit-keras

**Comment by wangermeng2021** [comment]
Thank you , I will check it
